src/main/java/jenkins/plugins/slack/StandardSlackService.java

```diff
@@ -207,6 +214,43 @@ boolean postToSlack(String apiEndpoint, JSONObject body) {
         return result;
     }
 
+    /**
+     * Make an HTTP POST upload to the Slack API
+     *
+     * @param apiEndpoint - The API endpoint to request, e.g. `files.upload`
+     * @param includes - includes comma-separated Ant-style globs as per {@link Util#createFileSet(File, String, String)} using {@code /} as a path separator;
+     *
+     * @return boolean indicating whether the API request succeeded
+     */
+    public boolean upload(FilePath workspace, String artifactIncludes, PrintStream log) {
+        boolean result = true;
+        if(workspace!=null) {
+            for(String roomId : roomIds) {
+                SlackFileRequest slackFileRequest = new SlackFileRequest(
+                workspace, populatedToken, roomId, null, artifactIncludes);
+                try {
+                    workspace.getChannel().callAsync(new SlackUploadFileRunner(log, Jenkins.get().proxy, slackFileRequest)).get();
+                } catch (IllegalStateException e) {
+                    logger.log(Level.WARNING, "IllegalStateException", e);
+                    result = false;
+                } catch (InterruptedException e) {
+                    logger.log(Level.WARNING, "InterruptedException", e);
```

Comment:
No need to log all of these separately, the exception message / stacktrace will say what it is, can you collapse it in to one catch block please